package/gluon-harden-dropbear/files/lib/gluon/upgrade/105-harden-dropbear

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/lua
+
+local uci = require('simple-uci').cursor()
+local util = require('gluon.util')
+
+
+local function is_root_pw_unlocked()
+	for line in io.lines("/etc/shadow") do
+		if line:match("^root:!") then
+			return false
+		end
+	end
+	return true
+end
+
+local function might_authorized_keys_exist()
+	local file = io.open("/etc/dropbear/authorized_keys", "r")
+	if not file then
+		return false
+	end
+	for line in file:lines() do
+		-- if the line is neither comments nor solely whitespaces
+		if not (line:match("^%s*#.*") or line:match("^%s*$")) then
```

Comment:
```suggestion
		if not (line:match("^%s*#") or line:match("^%s*$")) then
```
No need to match on arbitrary trailing characters.

Reply:
resolved in 4e85c03234f0ed20958b7824f4ab285ae1ca7329